src/main/java/sirius/biz/analytics/charts/explorer/PlatformDistributionTimeSeriesComputer.java

```diff
@@ -0,0 +1,84 @@
+/*
+ * Made with all the love in the world
+ * by scireum in Remshalden, Germany
+ *
+ * Copyright by scireum GmbH
+ * http://www.scireum.de - info@scireum.de
+ */
+
+package sirius.biz.analytics.charts.explorer;
+
+import sirius.biz.analytics.events.Event;
+import sirius.db.jdbc.SmartQuery;
+import sirius.kernel.commons.Strings;
+
+import javax.annotation.Nullable;
+import java.util.function.BiConsumer;
+
+/**
+ * Helps computing the platform distribution based on the user-agent data for a given Clickhouse event.
+ *
+ * @param <O> the type of entities expected by this computer
+ * @param <E> the type of events being queried
+ * @see sirius.biz.tycho.DashboardPlatformDistributionChart
+ */
+public class PlatformDistributionTimeSeriesComputer<O, E extends Event> extends EventTimeSeriesComputer<O, E> {
+    /**
+     * Retrieves the number of occurrences of the iOS platform within the user agent data.
+     */
+    public static final String AGGREGATION_EXPRESSION_IOS =
+            "countIf(position('Mac' in webData_userAgent) > 0 and position('Mobile' in webData_userAgent) > 0 or position('EdgiOS' in webData_userAgent) > 0)";
+
+    /**
+     * Retrieves the number of occurrences of the Android platform within the user agent data.
+     */
+    public static final String AGGREGATION_EXPRESSION_ANDROID =
+            "countIf(position('Android' in webData_userAgent) > 0 or position('EdgA' in webData_userAgent) > 0)";
+
+    /**
+     * Retrieves the number of occurrences of the Macintosh desktop platform within the user agent data.
+     */
+    public static final String AGGREGATION_EXPRESSION_MACINTOSH =
+            "countIf(position('Mac' in webData_userAgent) > 0 and position('Mobile' in webData_userAgent) == 0)";
```

Comment:
These constants repeat the same clause. Wasn't it better a method accepting a string?

Reply:
Maybe a `private static final String POSITION` for the position part and then a `Strings.apply()`.. might be good.

Reply:
The general idea of the static Strings is that no one feels tempted to touch or alter them. If we would have a method where we would assemble stuff, people might feel tempted to add a little bit over time